core/src/main/java/io/parsingdata/metal/data/ParseGraph.java

```diff
@@ -97,15 +100,15 @@ ParseGraph closeBranch() {
      * @return The first value (bottom-up) in this graph
      */
     public ParseValue current() {
-        if (isEmpty()) { return null; }
-        if (head.isValue()) {
-            return head.asValue();
-        }
-        if (head.isGraph()) {
-            final ParseValue value = head.asGraph().current();
-            if (value != null) { return value; }
-        }
-        return tail.current(); // Ignore current if it's a reference (or an empty graph)
+        return current(ImmutableList.create(this)).computeResult();
+    }
+
+    private SafeTrampoline<ParseValue> current(ImmutableList<ParseItem> items) {
+        if (items.isEmpty()) { return complete(() -> null); }
```

Comment:
See #161. There are some situations (especially with `ValueExpression`s, where the default return value is already an `Optional`) where it won't be straightforward to remove the use of `null`. But I think this one can actually quite easily be fixed :)